core/src/main/scala/org/http4s/multipart/MultipartParser.scala

```diff
@@ -591,6 +593,115 @@ object MultipartParser {
     go(stream, Stream.empty, 0)
   }
 
+  /////////////////////////////////////
+  // Resource-safe file-based parser //
+  /////////////////////////////////////
+
+  /** Like parseStreamedFile, but the produced parts' resources are managed by the supervisor.
+    */
+  def parseSupervisedFile[F[_]: Concurrent: Files](
```

Comment:
I think we could make these `private[multipart]` to give us more bincompat wiggle room in the future?

Reply:
Done.